### PR TITLE
[17.0][ADD] edi_project_oca: New module

### PR DESCRIPTION
**Pull request by SilvioC2C**

### CODE REVIEW

**Review by gurneyalex — APPROVED** [review]

File: edi_project_oca/models/project_project.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProjectProject(models.Model):
+    _name = "project.project"
+    _inherit = ["project.project", "edi.exchange.consumer.mixin"]
+
+    edi_disable_auto = fields.Boolean()
```
Comment (simahawk) [inline edi_project_oca/models/project_project.py]: no need to re-define this flag

File: edi_project_oca/models/project_project.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        projects = super().create(vals_list)
+        for project, vals in zip(projects, vals_list, strict=True):
+            project._event("on_project_create").notify(project, vals)
```
Comment (simahawk) [inline edi_project_oca/models/project_project.py]: no need to trigger these events. They are already available by default w/ component_event as `on_record_create` and `on_record_write`. 
What's the final goal for you?

File: edi_project_oca/models/project_task.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProjectTask(models.Model):
+    _name = "project.task"
+    _inherit = ["project.task", "edi.exchange.consumer.mixin"]
+
+    edi_disable_auto = fields.Boolean()
```
Comment (simahawk) [inline edi_project_oca/models/project_task.py]: same

File: edi_project_oca/models/project_task.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    @api.model_create_multi
+    def create(self, vals_list):
```
Comment (simahawk) [inline edi_project_oca/models/project_task.py]: same

File: edi_project_oca/readme/DESCRIPTION.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+This module intends to create a base to be extended by local EDI rules
+for project management.
+
+In order to add customizations for projects, create a listener:
+
+```python
+from odoo.addons.component.core import Component
+
+
+class ProjectEventListenerExample(Component):
```
Comment (simahawk) [inline edi_project_oca/readme/DESCRIPTION.md:10]: I don't understand why you need these listeners

**Review by simahawk — CHANGES_REQUESTED** [review]

File: edi_project_oca/models/project_project.py (hunk 1 above)
Comment (SilvioC2C) [reply edi_project_oca/models/project_project.py]: I used `oca_account_edi` as example to create this module, and the field is re-declared there for `account.move`, so I did the same for consistency

Comment (simahawk) [reply edi_project_oca/models/project_project.py]: that's wrong unless you want to control the visibility by the state

File: edi_project_oca/views/project_project.xml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                            <field name="exchange_record_count" />
+                        </span>
+                        <span class="o_stat_text">EDI</span>
+                    </div>
+                </button>
+            </xpath>
+            <xpath expr="//page[@name='settings']/group[last()]" position="inside">
+                <group
+                    name="edi_project_group"
+                    string="Electronic Data Interchange"
```
Comment (simahawk) [inline edi_project_oca/views/project_project.xml]:
```suggestion
                    string="EDI"
```

File: edi_project_oca/views/project_task.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                    <div class="o_form_field o_stat_info">
+                        <span class="o_stat_value">
+                            <field name="exchange_record_count" />
+                        </span>
+                        <span class="o_stat_text">EDI</span>
+                    </div>
+                </button>
+            </xpath>
+            <xpath expr="//page[@name='extra_info']/group[last()]" position="inside">
+                <group name="edi_project_group" string="Electronic Data Interchange">
```
Comment (simahawk) [inline edi_project_oca/views/project_task.xml]:
```suggestion
                <group name="edi_project_group" string="EDI">
```
Q: wouldn't be better to have a specific page for EDI? 
Just asking...

Comment (SilvioC2C) [reply edi_project_oca/views/project_task.xml]: Yep, can do that

File: edi_project_oca/tests/common.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+    @classmethod
+    def _build_and_add_component(cls, component_cls):
+        assert isinstance(component_cls, MetaComponent)
+        assert issubclass(component_cls, AbstractComponent)
+        component_cls._build_component(cls.comp_registry)
+        cls.comp_registry._cache.clear()
+
+    def _check_msg_in_log_output(self, msg: str, outputs: Iterable):
```
Comment (simahawk) [inline edi_project_oca/tests/common.py]: kind of complicating things? :smile: 
See here how you can collect dummy events https://github.com/OCA/edi/blob/14.0/edi_oca/tests/fake_components.py#L10
Yet... se below...

File: edi_project_oca/tests/test_project_listener.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+import logging
+
+from odoo.addons.component.core import Component
+
+from .common import TestEdiProjectOcaCommon
+
+_logger = logging.getLogger(__name__)
+
+
+class TestProjectListener(TestEdiProjectOcaCommon):
```
Comment (simahawk) [inline edi_project_oca/tests/test_project_listener.py]: you don't need all this testing... these are std events provided by component_event. You can simply drop them :wink:

**Review by simahawk — CHANGES_REQUESTED** [review]

**Review by simahawk — APPROVED** [review]

**Pull request merged**